ktranslate.wxs

```diff
@@ -1,489 +1,624 @@
-              <Component Id="cmp03C88EEF9204D5003FAB783161477DF7" Guid="*">
-                <File Id="filCC970C03CF92B1CD376E93C9E8EB3578" KeyPath="yes" Source="$(var.Path)/config/profiles/profiles/kentik_snmp/_general/ospf-mib.yml"/>
-              </Component>
-            </Directory>
-            <Directory Id="dirA53B1CD650FC07F57B1E067DB7CC4FF1" Name="nvidia">
-              <Component Id="cmp94EEE3A745C353AB113613460B65389E" Guid="*">
-                <File Id="fil0D457AA657B5A8BC2970CB21023CC209" KeyPath="yes" Source="$(var.Path)/config/profiles/profiles/kentik_snmp/nvidia/cumulus-linux-switch.yml"/>
-              </Component>
-            </Directory>
-            <Directory Id="dirB4A090D21EAD7A83AE7EB08D47BA4970" Name="opengear">
-              <Component Id="cmpA126ACB0F498F4DEE438895CBB146273" Guid="*">
-                <File Id="fil2A8EBA8B3A9096E1AEF8C1FB3C8C0EAC" KeyPath="yes" Source="$(var.Path)/config/profiles/profiles/kentik_snmp/opengear/console-manager.yml"/>
-              </Component>
-            </Directory>
-            <Directory Id="dirE2587F52A1C7B56C7E4DC6A617A360DE" Name="netapp">
-              <Component Id="cmp626A56F4B1F4A4283FBA8A5F0FAED47E" Guid="*">
-                <File Id="fil9542905B5C561ECAD1F3BB768316C9E4" KeyPath="yes" Source="$(var.Path)/config/profiles/profiles/kentik_snmp/netapp/netapp-cluster.yml"/>
-              </Component>
-            </Directory>
-            <Directory Id="dir7C3FE291B8A36BC431F1C99C49DCCB14" Name="synology">
-              <Component Id="cmp6EECEE2B0DE8DE8D1CF923543A167E0C" Guid="*">
-                <File Id="fil4B03CB9AE8F8FA944A073BBEAB26E9EC" KeyPath="yes" Source="$(var.Path)/config/profiles/profiles/kentik_snmp/synology/disk_station.yml"/>
-              </Component>
-            </Directory>
-            <Directory Id="dir8CDF592DD5D9903615A2F065ABEB4256" Name="eaton">
-              <Component Id="cmp800E9023057091384F8E8E0947CD881D" Guid="*">
-                <File Id="filA765332970E3BE5CACCF5449384B4DAE" KeyPath="yes" Source="$(var.Path)/config/profiles/profiles/kentik_snmp/eaton/traps.yml"/>
-              </Component>
-            </Directory>
-            <Directory Id="dirE36D30AFD3D9894CB9A7ABD05169C0A8" Name="isilon">
-              <Component Id="cmp354CC5099EA5863B101694798A000A65" Guid="*">
-                <File Id="fil2AA7DB19CC85A18876CA9C298949913F" KeyPath="yes" Source="$(var.Path)/config/profiles/profiles/kentik_snmp/isilon/isilon.yml"/>
-              </Component>
+        <DirectoryRef Id="APPLICATIONROOTDIRECTORY">
+            <Directory Id="dir4EEDBBDDE71B18DFA5755E20A41B3758" Name="config">
+                <Component Win64="$(var.Win64)" Id="cmpB46AF4C0CF9C68FA9C458C1219F7C1D1" Guid="*">
+                    <File Id="fil3A4553518059793557B7F50FFD0CF595" KeyPath="yes" Source="$(var.Path)/config/config.json"/>
+                </Component>
+                <Directory Id="dirF1204DAB15986CD6C35A715AFDAF37C5" Name="profiles">
```

Comment:
Is there a way to automatically discover these profile files? Right now the process is brittle because they are in another repo and get pulled in at build time with new files coming in all the time.

Reply:
Yes, it's possible to get something like this with the harvest tool from WiX https://wixtoolset.org/documentation/manual/v3/overview/heat.html
I implemented this in CI for snmp profiles